src/shared/exec-util.c

```diff
@@ -93,6 +93,10 @@ static int do_execute(
         bool parallel_execution;
         int r;
 
+        /* Make sure what we execute doesn't inherit NOTIFY_SOCKET from the parent process and tries to send
+         * notify messages to that socket. */
+        assert_se(unsetenv("NOTIFY_SOCKET") == 0);
```

Comment:
hmm, this should probably also unset LISTEN_FDS/LISTEN_FDNAMES/LISTEN_PID just in case.

And WATCHDOG_PID, WATCHDOG_USEC.

And SYSTEMD_EXEC_PID too.